hubstaff-export.rb

```diff
@@ -73,265 +320,255 @@ def run
     end
   end
 
+  def verbose?
+    options.verbose
+  end
+
   protected
 
-    def parsed_options?
-      # Specify options
-      @opts_parser = OptionParser.new do |opts|
-        opts.banner = "Usage: hubstaff-export COMMAND [OPTIONS]"
-        opts.separator  ""
-        opts.separator  "Commands"
-        opts.separator  "     authenticate app_token username password"
-        opts.separator  "       Authenticates and caches the tokens to 'hubstaff-client.cfg' in the current folder"
-        opts.separator  "     export-screens start_time stop_time"
-        opts.separator  "       Exports screenshots on a defined period."
-        opts.separator  "       Screenshots are exported into a folder structure like this"
-        opts.separator  "       - project - 34/user - 123/2015-07-01/123023-screen-0.jpg"
-        opts.separator  "       - (123023 is the hour, minutem second of the screenshot)"
-        opts.separator  "       Start and stop time must be in the ISO8601 format. e.g. YYYY-MM-DDThh:mmZ"
-        opts.separator  "       where Z means that the time is in UTC or it can be a timezone offset"
-        opts.separator  "       - ex.  2015-06-01T04:00Z  or 2015-06-01T00:00-0400 or 2015-06-01T05:00+0100"
-        opts.separator  "       - (those all represent the same time)"
-        opts.separator  ""
-        opts.separator  "Options"
-
-        opts.on('-v', '--version', 'version of the application')  { output_version ; exit 0 }
-        opts.on('-h', '--help', 'help method to show options')    { puts opts; exit 0 }
-        opts.on('-V', '--verbose', 'verbose the script calls')    { @options.verbose = true }
-
-        opts.on('-p', '--projects PROJECTS', 'comma separated list of project IDs')  {|projects| @options.projects = projects}
-        opts.on('-u', '--users USERS', 'comma separated list of user IDs')           {|users| @options.users = users}
-        opts.on(nil, '--no-ssl-verify', 'disable SSL certificate validation')        {|s| @options.skip_ssl_verify = s}
-
-        opts.on('-i', '--image_format IMAGE_EXPORT_TYPE', 'what image to export (full || thumb || both) (default is full only)') do |image_format|
-          @options.image_format = image_format
-        end
-        opts.on('-o', '--organizations ORGANIZATIONS', 'comma separated list of organization IDs (required)') do |organizations|
-          @options.organizations = organizations
-        end
-        opts.on('-d', '--directory DIRECTORY', 'a path to the output directory (otherwise ./screens is assumed)') do |directory|
-          @options.directory = directory
-        end
+  def parsed_options?
```

Comment:
We can use the `thor` gem to build this tool. It will help to reduce the amount of boilerplate code around the options parsing as well as eliminate the classes that inherit `BaseCommand`.

WDYT?

Reply:
Trying to not require any extra gems to keep the ease of use there. (users already had enough trouble installing ruby)